caikit/runtime/http_server/http_server.py

```diff
@@ -546,6 +590,64 @@ def _get_response_dataobject(rpc: CaikitRPCBase) -> Type[DataBase]:
         assert isinstance(dm_obj, type) and issubclass(dm_obj, DataBase)
         return dm_obj
 
+    @staticmethod
+    def _format_file_response(dm_class: Type[DataBase]) -> Response:
+        """Convert a dm_class into a fastapi file Response"""
+        file_obj = io.BytesIO()
+        file_info = dm_class.to_file(file_obj)
+
+        file_type = "application/octet-stream"
+        content_disposition = "attachment"
+        if file_info:
+            file_type = file_info.type if file_info.type else file_type
+            content_disposition = f'attachment; filename="{file_info.filename}"'
+
+        return Response(
+            content=file_obj.getvalue(),
+            headers={"Content-Disposition": content_disposition},
+            media_type=file_type,
+        )
+
+    @staticmethod
+    def _get_request_openapi(
+        pydantic_model: Union[pydantic.BaseModel, Type[pydantic.BaseModel]]
+    ):
+        """Helper to generate the openapi schema for a given request"""
+        raw_schema = pydantic_model.model_json_schema()
+        parsed_schema = flatten_json_schema(raw_schema)
+
+        multipart_schema = convert_json_schema_to_multipart(parsed_schema)
+
+        return {
+            "requestBody": {
```

Comment:
Cool! This is a _way_ better solution than the brutal hack I'd come up with a while ago